docs/source/release_notes.rst

```diff
@@ -3,6 +3,7 @@ Release Notes
 **Future Releases**
     * Enhancements
     * Fixes
+        * Text columns can be identified as a perfect match as an id column :pr:`4203`
```

Comment:
Might want to reword this to clarify, maybe something like "``IDColumnDataCheck`` now works with Unknown data type" or something